src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,296 @@
+     * @param wad The amount flushed out.
+     */
+    event Quit(address indexed quitTo, uint256 wad);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaInputConduit3/not-authorized");
+        _;
+    }
+
+    modifier isMate() {
+        require(may[msg.sender] == 1, "RwaInputConduit3/not-mate");
+        _;
+    }
+
+    /**
+     * @notice Define addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     * @param _to RwaUrn contract address.
+     */
+    constructor(address _psm, address _to) public {
+        require(_to != address(0), "RwaInputConduit3/invalid-to-address");
+
+        GemAbstract _gem = GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem());
+        psm = PsmAbstract(_psm);
+        dai = DaiAbstract(PsmAbstract(_psm).dai());
+        gem = _gem;
+        to = _to;
+
+        uint256 gemDecimals = _gem.decimals();
+        uint256 daiDecimals = dai.decimals();
+        require(gemDecimals <= daiDecimals, "RwaInputConduit3/invalid-gem-decimals");
+        to18ConvertionFactor = 10**(daiDecimals - gemDecimals);
+
+        // Give unlimited approve to PSM gemjoin
+        _gem.approve(address(PsmAbstract(_psm).gemJoin()), type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo", "to"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            require(data != address(0), "RwaInputConduit3/invalid-quit-to-address");
+            quitTo = data;
+        } else if (what == "to") {
+            require(data != address(0), "RwaInputConduit3/invalid-to-address");
+            to = data;
```

Comment:
why is `to` fileable as opposed to RwaInputConduit2?

Reply:
The input conduit address is going to be provided to the crypto broker and will be the address to which they need to send Dai for principal repayment.
If we ever need to update this, it's a very cumbersome process, because it requires changes in the legal documents.
We added this as an escape hatch to allow governance to update the underlying `RwaUrn` address for any reason (i.e.: a bug is found) without having to change the address of the input conduit.

Reply:
Isn't what we have `quitTo` for?

Reply:
I see it is for `isMate`, but I guess governance can add itself and still use it.

Reply:
`quitTo` is a solution for a different issue.
In case the USDC cannot be converted to DAI using the PSM, Maker governance needs to be able to send USDC back to the crypto broker after we figure out a new route for payment.

Reply:
`quitTo` is meant to solve a different issue.
If the crypto broker sends the wrong amount of gem to this conduit, they can claim it back to the `quitTo` address without Maker governance involvement.

Reply:
Yes but that doesn't change the fact that governance can still use it as an escape path and we avoid differing even more from the other versions.
I still feel these all paths could maybe be more homogenous but need to think a bit more about it.

Reply:
The multiplication of the number of paths derive from the fact that now we are dealing with 2 different assets, so we need escape hatches for both.

If the escape hatch needs to be used because of some operational error (i.e.: someone fat-fingering a `0`), there is no need to involve Maker governance, hence the `quit()` method. Governance is only involved if we ever need to change the destination address for `quit()`.

If it's something more difficult to tackle (i.e.: the PSM being empty), then we do need Governance to interfere and decide what to do: should it burn the outstanding Dai and cancel out the loan? Should it simply send Dai instead of USDC to the crypto broker?

Reply:
Resolving it as I understand `to` fileable as a means for updating a new urn without redeploying the conduit.